lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -1429,12 +1429,18 @@ bool HandshakeBuilder::visitHandshake(MemoryOp op) {
   // Create the special type to represent this memory.
   FIRRTLType memType = MemOp::getTypeForPortList(depth, dataType, ports);
```

Comment:
Should getTypeForPortList return a BundleType?  That would eliminate the cast on 1434

Reply:
Yes it should, I did this in 1d65914, so you can remove the `cast<BundleType>` below.

Reply:
Sweet. Thanks! Will be fixed in the rebase.

Reply:
There is one weird edge case with this change, specifically if you have a write-only memory (https://github.com/llvm/circt/issues/492). In this situation, canonicalization will try to push the internal flip out and the port type can't be cast to `BundleType`. 

And yes, I do admit the lack of plausibility of a write-only memory...